src/tracker/psmove_tracker.c

```diff
@@ -56,41 +56,51 @@
 #endif
 
 //#define DEBUG_WINDOWS 			// shall additional windows be shown
-#define ROIS 4                   	// the number of levels of regions of interest (roi)
-#define BLINKS 2                 	// number of diff images to create during calibration
-#define BLINK_DELAY 200             	// number of milliseconds to wait between a blink
-#define CALIB_MIN_SIZE 50		 	// minimum size of the estimated glowing sphere during calibration process (in pixel)
-#define CALIB_SIZE_STD 10	     	// maximum standard deviation (in %) of the glowing spheres found during calibration process
-#define CALIB_MAX_DIST 30		 	// maximum displacement of the separate found blobs
-#define COLOR_FILTER_RANGE_H 20		// +- H-Range of the hsv-colorfilter
-#define COLOR_FILTER_RANGE_S 85		// +- s-Range of the hsv-colorfilter
-#define COLOR_FILTER_RANGE_V 85		// +- v-Range of the hsv-colorfilter
-
-/* Thresholds */
-#define ROI_ADJUST_FPS_T 160		// the minimum fps to be reached, if a better roi-center adjusment is to be perfomred
-#define CALIBRATION_DIFF_T 20		// during calibration, all grey values in the diff image below this value are set to black
-// if tracker thresholds not met, sphere is deemed not to be found
-#define TRACKER_QUALITY_T1 0.3f		// minimum ratio of number of pixels in blob vs pixel of estimated circle.
-#define TRACKER_QUALITY_T2 0.7f		// maximum allowed change of the radius in percent, compared to the last estimated radius
-#define TRACKER_QUALITY_T3 4.0f		// minimum radius
-#define TRACKER_ADAPTIVE_XY 1		// specifies to use a adaptive x/y smoothing
-#define TRACKER_ADAPTIVE_Z 1		// specifies to use a adaptive z smoothing
-#define COLOR_ADAPTION_QUALITY 35 	// maximal distance (calculated by 'psmove_tracker_hsvcolor_diff') between the first estimated color and the newly estimated
-#define COLOR_UPDATE_RATE 1	 	 	// every x seconds adapt to the color, 0 means no adaption
-// if color thresholds not met, color is not adapted
-#define COLOR_UPDATE_QUALITY_T1 0.8f	// minimum ratio of number of pixels in blob vs pixel of estimated circle.
-#define COLOR_UPDATE_QUALITY_T2 0.2f	// maximum allowed change of the radius in percent, compared to the last estimated radius
-#define COLOR_UPDATE_QUALITY_T3 6.0f	// minimum radius
-
+#define ROIS 4                          // the number of levels of regions of interest (roi)
+#define BLINKS 2                        // number of diff images to create during calibration
+#define COLOR_MAPPING_RING_BUFFER_SIZE 256  /* Has to be 256, so that next_slot automatically wraps */
 #define PSEYE_BACKUP_FILE "PSEye_backup.ini"
-
-#define INTRINSICS_XML "intrinsics.xml"
-#define DISTORTION_XML "distortion.xml"
-
 #define COLOR_MAPPING_DAT "colormapping.dat"
 
-/* Only re-use color mappings "younger" than 2 hours */
-#define COLOR_MAPPING_MAX_AGE (2*60*60)
+static const PSMoveTrackerSettings tracker_default_settings = {
```

Comment:
Thanks, but I can't take credit for it, this was either @cboulay or @brendanwalker :)